DxDispatch/src/dxdispatch/dxDispatchWrapper.cpp

```diff
@@ -0,0 +1,304 @@
+#include "pch.h"
+#include "Adapter.h"
+#include "Device.h"
+#include "Model.h"
+#include "Dispatchable.h"
+#include "JsonParsers.h"
+#include "Executor.h"
+#include "CommandLineArgs.h"
+#include "ModuleInfo.h"
+#include "dxDispatchWrapper.h"
+
+using namespace Microsoft::WRL;
+
+class ModelWrapper
+{
+public:
+    ModelWrapper(Model&& model):
+        m_model(std::move(model))
+    {}
+
+    ~ModelWrapper() = default;
+    Model &Value()
+    {
+        return m_model;
+    }
+private:
+    Model m_model;
+};
+
+HRESULT DxDispatch::CreateDxDispatchFromJsonString(
+    _In_            int argc,
+    _In_            char** argv,
+    _In_opt_        LPCSTR jsonConfig,
+    _In_opt_        IUnknown *adapterUnk,
+    _In_opt_        IDxDispatchLogger *customLogger,
+    _COM_Outptr_    IDxDispatch **dxDispatch)
+{
+    ComPtr<DxDispatch> dxDispatchImpl;
+    ComPtr<IAdapter> adapter;
+
+    if(adapterUnk)
+    {
+        RETURN_IF_FAILED(adapterUnk->QueryInterface(IID_PPV_ARGS(&adapter)));
+    }
+    dxDispatchImpl = new DxDispatch();
+    RETURN_IF_FAILED(dxDispatchImpl->Intialize(argc, argv, jsonConfig, adapter.Get(), customLogger));
+    return dxDispatchImpl->QueryInterface(IID_PPV_ARGS(dxDispatch));
+}
+
+HRESULT DxDispatch::QueryInterface( 
+    REFIID riid,
+    _COM_Outptr_ void** ppvObject)
+{
+    if((IID_IUnknown ==  riid) || 
+        __uuidof(IDxDispatch) == riid)
+    {
+        *ppvObject = static_cast<IDxDispatch*>(this);
+    }
+    else
+    {
+        return E_NOTIMPL;
+    }
+    AddRef();
+    return S_OK;
+} 
+
+ULONG DxDispatch::AddRef( void)
+{
+    return ++m_refCount;
+}
+
+ULONG DxDispatch::Release( void)
+{
+    auto ref = --m_refCount;
+    if(ref == 0)
+    {
+        delete this;
+    }
+    return ref;
+}
+
+HRESULT DxDispatch::RunAll() try
+{
+    if(m_modelWrapper)
+    {
+        RETURN_IF_FAILED(m_pixCaptureHelper->BeginCapturableWork());
+        Executor executor{m_modelWrapper->Value(), m_device, *m_options, m_logger.Get()};
+        executor.Run();
+        RETURN_IF_FAILED(m_pixCaptureHelper->EndCapturableWork());
+        return S_OK;
+    }
+    else
+    {
+        RETURN_IF_FAILED(E_UNEXPECTED);
+    }
+    
+} CATCH_RETURN();
+
+UINT32 DxDispatch::GetCommandCount()
+{
+    return m_commandCount;
+}
+
+HRESULT DxDispatch::RunCommand(
+            UINT32 index) try
+{
+    if(index != m_currentIndex)
+    {
+        return E_INVALIDARG;
+    }
+    return E_NOTIMPL;
```

Comment:
TODO in a future change? If this isn't something you see being used in the near term, maybe just remove the API?